app/src/main/java/ru/prometeydev/movie/model/network/MoviesApiProvider.kt

```diff
@@ -0,0 +1,68 @@
+package ru.prometeydev.movie.model.network
+
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.Response
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+import retrofit2.create
+import ru.prometeydev.movie.BuildConfig
+import java.util.concurrent.TimeUnit
+
+class MoviesApiProvider {
+
+    lateinit var api: MoviesApi
+
+    private val okHttpClient: OkHttpClient
+
+    init {
+        val tokenInterceptor = TokenInterceptor()
```

Comment:
Mental - try to keep constructors really light weight. This is heavy and can crash with some exception. Hard to test and mind blowing what to do with crash in runtime. Also possible performance penalty.